### DISCUSSION TITLE
Explicit prop types over flags

### DISCUSSION THREAD

**Discussion by aminpaks** [Shopify/polaris-react#6933] [Ideas] [answered]
Hi team,

I was wondering if you guys have considered to opt-out of flag option props for component in favour of explicit props.
For example we got the Button component that accepts `plain`, and `outline` as flags, and it breaks when we combine the two but it wouldn’t break if the prop was `variant="plain"` or `variant="outline"`.

**Comment by robphoenix** [comment, marked as answer]
I believe this is now the approach being taken? https://github.com/Shopify/polaris/pull/10090

**Reply by aminpaks** [reply]
Correct!
